### PR TITLE
Fixing the loading and logging of History class

### PR DESCRIPTION
**Pull request by Bev-low**
There was a bug becasue of the following reasons 
- using LocalDateTime instead of LocalDate
- Some parts were using yyyy-MM-dd and some were using dd/MM/yyyy
- There was an additional wrapper for history when it loads so like within the data. History had another History inside, so the load was reading Null. The additonal wrapper was caused by hisotry being a hashmap. 
To solve that, I toJson each day instead of the hisotyr itself. This can easily be changed to log (which is a weapper for day, meals, water). 

Now History loads and logs properly

### CODE REVIEW

File: src/main/java/core/DateSerializer.java [hunk 1]
```diff
@@ -6,20 +6,21 @@
...
-    public JsonElement serialize(LocalDateTime src, Type typeOfSrc, JsonSerializationContext context) {
+    public JsonElement serialize(LocalDate src, Type typeOfSrc, JsonSerializationContext context) {
         return new JsonPrimitive(src.format(formatter));  
     }
 
     @Override
-    public LocalDateTime deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) {
-        return LocalDateTime.parse(json.getAsString(), formatter); 
+    public LocalDate deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) {
+        System.out.println("enter deseralize");
```
Comment (TVageesan) [inline src/main/java/core/DateSerializer.java:23]: Make sure to delete debug logs before merging to master - a habit I like to use is comment //DEBUG next to it and ctrl + shift + f checking for it before committing

File: src/main/java/core/History.java [hunk 2]
```diff
@@ -2,42 +2,65 @@
...
-        return gson.fromJson(jsonObject, History.class);
+        History history = new History();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+
+        // Iterate through the JSON keys (dates) and deserialize them as LocalDate and Day objects
+        for (Map.Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+            LocalDate date = LocalDate.parse(entry.getKey(), formatter);  // Convert key to LocalDate
+            System.out.println(date);
+            Day day = gson.fromJson(entry.getValue(), Day.class);  // Deserialize the Day object
+            System.out.println(day);
```
Comment (TVageesan) [inline src/main/java/core/History.java:57]: Same issue with debug logs

**Review by TVageesan — APPROVED** [review]
Looks good logic wise to me, feel free to merge after resolving checkstyle + removing debug logs

File: src/main/java/core/DateSerializer.java (hunk 1 above)
Comment (Bev-low) [reply src/main/java/core/DateSerializer.java]: okay can tahnks, just removed!

**Pull request merged**